mkt/webpay/urls.py

```diff
@@ -22,7 +22,8 @@
         name='webpay-prepare'),
     url(r'^webpay/inapp/prepare/', PreparePayInAppView.as_view(),
         name='webpay-prepare-inapp'),
-    url(r'^webpay/failure/(?P<pk>\d+)/', FailureNotificationView.as_view(),
+    url(r'^webpay/failure/(?P<pk>(?:simulate|free|webpay)?:?[a-z0-9-]+)/',
```

Comment:
This can be simplified to be more like the other patterns that match on everything that isn't a '/'